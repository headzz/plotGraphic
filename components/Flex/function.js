export const justifyContent = (align = "flex-start") => {
  return `justify-content: ${align}`;
};

export const checkAlign = (props) => {
  if (props.center) {
    return justifyContent("center");
  }

  if (props.right) {
    return justifyContent("flex-end");
  }

  if (props.spaceBetween) {
    return justifyContent("space-between");
  }

  return justifyContent();
};
