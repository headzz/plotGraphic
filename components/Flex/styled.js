import styled from "styled-components";
import { checkAlign } from "./function";

export const FlexStyled = styled.div`
  display: flex;
  ${(props) => checkAlign(props)};
  flex-direction: row;
  width: 100%;
`;
