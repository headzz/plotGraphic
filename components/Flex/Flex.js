import { FlexStyled } from "./styled";
import PropTypes from "prop-types";
import React from "react";

/**
 * @component
 * @name Flex
 * @description Gives flex display for components.
 * @namespace components components/Body
 * @param {boolean} center Justify content for center.
 * @param {element} children Component(s) that may appear inside the layout.
 * @param {boolean} right Justify content for flex-end.
 * @param {boolean} spaceBetween Justify content for space-between.
 *
 */

export const Flex = ({ center, children, right, spaceBetween }) => {
  return (
    <FlexStyled
      data-testid="flex"
      center={center}
      right={right}
      spaceBetween={spaceBetween}
    >
      {children}
    </FlexStyled>
  );
};

Flex.propTypes = {
  center: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  right: PropTypes.bool,
  spaceBetween: PropTypes.bool,
};

Flex.defaultProps = {
  center: false,
  right: false,
  spaceBetween: false,
};
