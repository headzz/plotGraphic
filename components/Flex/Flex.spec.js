import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { Flex } from "./Flex";
afterEach(cleanup);

test("render flex content", () => {
  const { container, getByTestId } = render(
    <Flex>
      <div>1</div>
      <div>2</div>
    </Flex>
  );
  expect(getByTestId("flex")).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});

test("render flex-content center", () => {
  const { container, getByTestId } = render(
    <Flex center>
      <div>1</div>
      <div>2</div>
    </Flex>
  );
  expect(getByTestId("flex")).toHaveStyle("justify-content: center");
  expect(container.firstChild).toMatchSnapshot();
});

test("render flex-content right", () => {
  const { container, getByTestId } = render(
    <Flex right>
      <div>1</div>
      <div>2</div>
    </Flex>
  );
  expect(getByTestId("flex")).toHaveStyle("justify-content: flex-end");
  expect(container.firstChild).toMatchSnapshot();
});

test("render flex-content spaceBetween", () => {
  const { container, getByTestId } = render(
    <Flex spaceBetween>
      <div>1</div>
      <div>2</div>
    </Flex>
  );
  expect(getByTestId("flex")).toHaveStyle("justify-content: space-between");
  expect(container.firstChild).toMatchSnapshot();
});
