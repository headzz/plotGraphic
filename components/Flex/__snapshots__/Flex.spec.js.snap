// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render flex content 1`] = `
<div
  class="sc-bdVaJa gRiLhU"
  data-testid="flex"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;

exports[`render flex-content center 1`] = `
<div
  class="sc-bdVaJa hAvHhV"
  data-testid="flex"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;

exports[`render flex-content right 1`] = `
<div
  class="sc-bdVaJa gXLsGA"
  data-testid="flex"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;

exports[`render flex-content spaceBetween 1`] = `
<div
  class="sc-bdVaJa cqvdvB"
  data-testid="flex"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;
